src/scheduler.cc

```diff
@@ -225,18 +225,17 @@ Status SchedulerService::TaskInfo(ServerContext* context, const TaskInfoRequest*
   auto failed_tasks = failed_tasks_.get();
   auto computation_graph = computation_graph_.get();
   auto workers = workers_.get();
-  for (int i = 0; i < failed_tasks->size(); ++i) {
-    TaskStatus* info = reply->add_failed_task();
-    *info = (*failed_tasks)[i];
+  for (auto i = failed_tasks->begin(); i != failed_tasks->end(); ++i) {
```

Comment:
Both conventions are used (e.g. [search for `iterator i =` here](http://www.boost.org/doc/libs/1_58_0/boost/container/map.hpp)); I used to do that and got tired of it a few years ago.